rsmt2d_test.go

```diff
@@ -157,3 +158,57 @@ func TestEdsRepairTwice(t *testing.T) {
 		})
 	}
 }
+
+// TestRepairWithOneQuarterPopulated is motivated by a use case from
+// celestia-node. It verifies that a new EDS can be populated via SetCell. After
+// enough chunks have been populated, it verifies that the EDS can be repaired.
+// After the EDS is repaired, the test verifies that data in a repaired cell
+// matches the expected data.
+func TestRepairWithOneQuarterPopulated(t *testing.T) {
+	edsWidth := 4
+	chunkSize := 512
+
+	exampleEds := createExampleEds(t, chunkSize)
+
+	eds, err := rsmt2d.NewExtendedDataSquare(rsmt2d.NewLeoRSCodec(), rsmt2d.NewDefaultTree, uint(edsWidth), uint(chunkSize))
+	require.NoError(t, err)
+
+	// Populate EDS with 1/4 of chunks using SetCell
+	err = eds.SetCell(0, 0, exampleEds.GetCell(0, 0))
+	require.NoError(t, err)
+	err = eds.SetCell(0, 1, exampleEds.GetCell(0, 1))
+	require.NoError(t, err)
+	err = eds.SetCell(1, 0, exampleEds.GetCell(1, 0))
+	require.NoError(t, err)
+	err = eds.SetCell(1, 1, exampleEds.GetCell(1, 1))
+	require.NoError(t, err)
```

Comment:
Considered refactoring to

```go
	for row := 0; row < edsWidth/2; row++ {
		for col := 0; col < edsWidth/2; col++ {
			err = eds.SetCell(uint(row), uint(col), exampleEds.GetCell(uint(row), uint(col)))
			require.NoError(t, err)
		}
	}
```

but I think the existing implementation is easier to read.